pkg/ui/query.go

```diff
@@ -65,38 +64,32 @@ func queryTmplFuncs() template.FuncMap {
 
 // Register registers new GET routes for subpages and redirects from / to /graph.
 func (q *Query) Register(r *route.Router, ins extpromhttp.InstrumentationMiddleware) {
-	instrf := func(name string, next func(w http.ResponseWriter, r *http.Request)) http.HandlerFunc {
-		return ins.NewHandler(name, http.HandlerFunc(next))
-	}
+	r.Get("/", func(w http.ResponseWriter, r *http.Request) {
+		http.Redirect(w, r, path.Join(GetWebPrefix(q.logger, q.externalPrefix, q.prefixHeader, r), "/graph"), http.StatusFound)
+	})
 
-	r.Get("/", instrf("root", q.root))
-	r.Get("/graph", instrf("graph", q.graph))
-	r.Get("/stores", instrf("stores", q.stores))
-	r.Get("/status", instrf("status", q.status))
-
-	r.Get("/static/*filepath", instrf("static", q.serveStaticAsset))
-	// Make sure that "<path-prefix>/new" is redirected to "<path-prefix>/new/" and
-	// not just the naked "/new/", which would be the default behavior of the router
-	// with the "RedirectTrailingSlash" option (https://godoc.org/github.com/julienschmidt/httprouter#Router.RedirectTrailingSlash),
-	// and which breaks users with a --web.route-prefix that deviates from the path derived
-	// from the external URL.
-	r.Get("/new", func(w http.ResponseWriter, r *http.Request) {
-		http.Redirect(w, r, path.Join(GetWebPrefix(q.logger, q.externalPrefix, q.prefixHeader, r), "new")+"/graph", http.StatusFound)
+	r.Get("/classic/", func(w http.ResponseWriter, r *http.Request) {
+		http.Redirect(w, r, path.Join(GetWebPrefix(q.logger, q.externalPrefix, q.prefixHeader, r), "/classic/graph"), http.StatusFound)
 	})
-	r.Get("/new/*filepath", instrf("react-static", q.serveReactUI))
+
+	// Redirect the original React UI's path (under "/new") to its new path at the root.
+	r.Get("/new/*path", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Why we need this? There are some hardcoded React paths for /new?

Reply:
well nothing related to the code itself it's more because the user would have in the browser history the url prefixed by `/new`.

So it's just to avoid they will have an error by autocompleting the url in their browser

Reply:
ah. nice!